testsuite/MDAnalysisTests/coordinates/test_reader_api.py

```diff
@@ -131,56 +132,75 @@ def _check_slice(self, start, stop, step):
         ref = self.reference[start:stop:step]
 
         assert_equal(res, ref)
-    
+
     def test_slices(self):
         for start, stop, step in [
-                (None, None, None),  # blank slice
-                (None, 5, None),  # set end point
-                (2, None, None),  # set start point
-                (2, 5, None),  # start & end
-                (None, None, 2),  # set skip
-                (None, None, -1),  # backwards skip
-                (0, 10, 1),
-                (0, 10, 2),
-                (None, 20, None),  # end beyond real end
-                (None, 20, 2),  # with skip
-                (0, 5, 2),
-                (5, None, -1),
-                (None, 5, -1),
+            (None, None, None),  # blank slice
+            (None, 5, None),  # set end point
+            (2, None, None),  # set start point
+            (2, 5, None),  # start & end
+            (None, None, 2),  # set skip
+            (None, None, -1),  # backwards skip
+            (0, 10, 1),
+            (0, 10, 2),
+            (None, 20, None),  # end beyond real end
+            (None, 20, 2),  # with skip
+            (0, 5, 2),
+            (5, None, -1),
+            (None, 5, -1),
+            (100, 10, 1),
+            (-10, None, 1),
+            (100, None, -1),  # beyond real end
+            (100, 5, -20),
+            # (-5, -50, -1)
         ]:
             yield self._check_slice, start, stop, step
 
     def test_slice_IE_1a(self):
         """Stop less than start"""
+
         def sl():
             return list(self.reader[5:1:1])
-        assert_raises(IndexError, sl)
+
+        assert_equal(self.reference[5:1:1], sl)
 
     def test_slice_IE_1b(self):
         """Stop less than start"""
+
         def sl():
             return list(self.reader[1:5:-1])
-        assert_raises(IndexError, sl)
+
+        assert_equal(self.reference[1:5:-1], sl)
 
     def test_slice_IE_2(self):
         """Outside of range of trajectory"""
+
         def sl():
             return list(self.reader[100:])
-        assert_raises(IndexError, sl)
+
+        assert_equal(self.reference[100:], sl)
 
     def test_slice_IE_3(self):
         def sl():
             return list(self.reader[-100:])
-        assert_raises(IndexError, sl)
+
+        ref = self.reference[-100:]
+        count = 0
+        for i in sl():
```

Comment:
Also, the point of having the `sl` sub-function was to test the exception. There is no need for such a complicated system anymore; comparing directly `list(self.reader[-100:])` with `self.reference[-100:]` should do just as well. The same goes for some of the tests above.